lucille-core/src/test/java/com/kmwllc/lucille/stage/DropDocumentTest.java

```diff
@@ -119,4 +119,46 @@ public void testNumberConditional() throws StageException {
     stage.processConditional(doc3);
     assertTrue(doc3.isDropped());
   }
+
+  @Test
+  public void testDropWithFiftyPercent() throws StageException {
+    Stage stage = factory.get("DropDocumentTest/p50.conf");
+
+    int n = 100;
+    int dropped = 0;
+    int kept = 0;
+
+    for (int i = 0; i < n; i++) {
+      Document doc = Document.create("doc" + i);
+      stage.processDocument(doc);
+
+      if (doc.isDropped()) {
+        dropped++;
+      } else {
+        kept++;
+      }
+    }
+
+    assertTrue(dropped > 0);
+    assertTrue(kept > 0);
```

Comment:
This test isn't deterministic and there's a chance it can fail despite the Stage working properly... there's a chance, albeit very very small (7.8886091e-31), that we could drop or keep all of the documents. 

This may be ok since the chances are so small but want to call it out.

Reply:
We might want to just check that dropped + kept == 100 instead... that is guaranteed to be true